src/main/kotlin/uk/gov/justice/digital/hmpps/pecs/jpc/location/importer/LocationsSpreadsheet.kt

```diff
@@ -20,9 +20,8 @@ private const val ROW_OFFSET = 1
 class LocationsSpreadsheet(private val spreadsheet: Workbook, private val locationRepository: LocationRepository) : Closeable {
 
     init {
-        Tab.values().toList().filter { spreadsheet.getSheet(it.label) == null }.joinToString { it.label }.let {
-            if (it.isNotBlank()) throw IllegalStateException("The following tabs are missing from the locations spreadsheet: $it")
-        }
+        Tab.values().toList().filter { spreadsheet.getSheet(it.label) == null }.joinToString { it.label }
+                .takeIf { it.isNotBlank() }?.let { throw NullPointerException("The following tabs are missing from the locations spreadsheet: $it") }
```

Comment:
sanity check; should not be it.isBlank? i.e throw exception only when blank?

Reply:
No. if the joined string is not blank then we have some tabs that are not present.